source/grid/grid_tools.cc

```diff
@@ -2103,6 +2103,59 @@ namespace GridTools
     return output_tuple;
   }
 
+  template <int spacedim>
+#ifndef DOXYGEN
+  std::tuple<std::map<unsigned int, std::vector<unsigned int>>,
+             std::map<unsigned int, unsigned int>,
+             std::map<unsigned int, std::vector<unsigned int>>>
+#else
+  return_type
+#endif
+  guess_point_owner(
+    const RTree<std::pair<BoundingBox<spacedim>, unsigned int>> &covering_rtree,
+    const std::vector<Point<spacedim>> &                         points)
+  {
+    std::map<unsigned int, std::vector<unsigned int>> point_owners;
+    std::map<unsigned int, unsigned int>              map_owners_found;
+    std::map<unsigned int, std::vector<unsigned int>> map_owners_guessed;
+    std::vector<std::pair<BoundingBox<spacedim>, unsigned int>> search_result;
+
+    unsigned int n_points = points.size();
+    for (unsigned int pt = 0; pt < n_points; ++pt)
+      {
+        search_result.clear(); // clearing last output
+
+        // Running tree search
+        covering_rtree.query(boost::geometry::index::intersects(points[pt]),
+                             std::back_inserter(search_result));
+
+        // Keep track of how many processes we guess to own the point
+        std::set<unsigned int> owners_found;
+        // Check in which other processes the point might be
+        for (const auto &box_rk : search_result)
```

Comment:
It took me longer than I would like to admit to figure out that `box_rk` is short for 'rank bounding box'. Could we instead name this `rank_bbox` to make it more obvious?